src/main/groovy/edu/oregonstate/mist/locations/frontend/resources/LocationResource.groovy

```diff
@@ -82,21 +84,109 @@ class LocationResource extends Resource {
         }
 
         String result = locationDAO.search(trimmedQ, trimmedCampus, trimmedType, pageNumber, pageSize)
-        ok(result).build()
+
+        ResultObject resultObject = new ResultObject()
+        resultObject.data = []
```

Comment:
data may be a list of a single object. When the user calls /locations - data holds a list and when the user calls /locations/123abc - data holds a single object.